File: cellab_devkit-1.0 (1)/rules_java/perfumer.java
/*

    Diffusion of a random gas

    No lookup table is used.

*/

class Perfumer extends ruletable {
    void jcruleModes() {
        setWorld(13);                   // Own code, torus world
        setOwnCodeRequest("gaswall");   // User own code function
        setPatternRequest("perfume");
        setPaletteRequest("perfume");
    }

    int jcrule(int oldstate) {
        return 0;               // No lookup table
    }
}

public class perfumer {
    public static void main(String args[]) {
        (new Perfumer()).generateRuleFile("perfumer");
    }
}

File: references/cellab_devkit-1.0 (1)/ruledefs/perfumer.js

    /*
            
                Diffusion of a random gas
                
    */
    
    rule.worldtype = 13;         // 2D torus world, user evaluator
    rule.patreq = "perfume";
    rule.palreq = "perfume";
    rule.ocodereq = "gaswall";

    function perfumer(oldstate,     nw, n  , ne,
                                    w, self, e,
                                    sw, s  , se) {
        return 0;
    }
